### DISCUSSION TITLE
Set up of roi_size of sliding_window_inference, and spatial_size in  RandCropByPosNegLabeld for 3D dataset with images of different Z dimension.

### DISCUSSION THREAD

**Discussion by LalehSeyyed** [Project-MONAI/MONAI#3093] [Q&A]
I work on CT scan 3D images with dimensions (512,512, Z) where Z is different for each case, ranging from 46 to 400. In such cases how we should set the roi_size in sliding_window_inference? 

Also, for RandCropByPosNegLabeld, the largest spatial_size that fit in my structure considering the network and image sizes was (256,256,16). It is not clear for setting this parameter it is better to go with the largest possible size or the smaller one? 

Can someone guide me for the logic behind choosing the best setting for these parameters

**Comment by Nic-Ma** [comment]
Hi @LalehSeyyed ,

Thanks for raising this interesting topic.
Actually, I think there is no concrete solution to choose the sliding window side, it depends on your dataset content and valid image region in it, need to analyze with some experience.
@dongyang0122 May help share more details.

Thanks.

**Comment by LalehSeyyed** [comment]
Thanks.  Among the different slices in z-direction which are ranging from 46 to 400, only a few images (e.g 10%) may have the label of interest (e.g Tumor). Please let me know if I need to provide any other information so that you can guide me through this. I will asl add @ericspod, @wyli, and @rijobro. Any tips or logic behind tuning these parameters are appreciated.

**Comment by dongyang0122** [comment]
Hi @LalehSeyyed,

The ROI size in random cropping or the sliding-window size depends on the applications. If the application requires more global context (e.g., organ segmentation), you may want to increase the window size (e.g., 512 x 512 x 512) to avoid outliers; if the application focuses more on local appearance (e.g., tumor segmentation), smaller window size (e.g., 128 x 128 x 128) would be a better choice.

Best,
Dong
